src/core/execute.c

```diff
@@ -1156,6 +1160,29 @@ void exec_context_dump(const ExecContext *c, FILE* f, const char *prefix) {
                 prefix, proc_subset_to_string(c->proc_subset),
                 prefix, private_bpf_to_string(c->private_bpf));
 
+        if (c->private_bpf == PRIVATE_BPF_YES) {
+                _cleanup_free_ char *commands = bpf_delegate_commands_to_string(c->bpf_delegate_commands);
+                if (!commands)
+                        return (void) log_oom();
+
+                _cleanup_free_ char *maps = bpf_delegate_maps_to_string(c->bpf_delegate_maps);
+                if (!maps)
+                        return (void) log_oom();
+
+                _cleanup_free_ char *programs = bpf_delegate_programs_to_string(c->bpf_delegate_programs);
+                if (!programs)
+                        return (void) log_oom();
+
+                _cleanup_free_ char *attachments = bpf_delegate_attachments_to_string(c->bpf_delegate_attachments);
+                if (!attachments)
+                        return (void) log_oom();
+
+                fprintf(f, "%sBPFDelegateCommands: %s\n", prefix, commands);
+                fprintf(f, "%sBPFDelegateMaps: %s\n", prefix, maps);
+                fprintf(f, "%sBPFDelegatePrograms: %s\n", prefix, programs);
+                fprintf(f, "%sBPFDelegateAttachments: %s\n", prefix, attachments);
```

Comment:
Then, if _to_string() failed, `commands` and friends may be NULL. So please wrap them with `strna()`.